### ISSUE TITLE
mirrord progress does not print warnings, output gets messed when added

### ISSUE THREAD

**Issue by infiniteregrets** [metalbear-co/mirrord#1664]
check commits in this PR: https://github.com/metalbear-co/mirrord/pull/1654
```
mehularora@mehuls-air mirrord % target/debug/mirrord exec -t deploy/py-serv-deployment  curl ifconfig.me
⠧ mirrord cli starting...
  ⠧ exec
⠇ mirrord cli starting...
  ⠇ exec
⠏ mirrord cli starting...
  ⠏ exec
    ⠧ preparing to launch process
⠋ mirrord cli starting...
  ⠋ exec
    ⠇ preparing to launch process
⠙ mirrord cli starting...
  ⠙ exec
    ⠏ preparing to launch process
⠹ mirrord cli starting...
  ⠹ exec
    ⠋ preparing to launch process
⠸ mirrord cli starting...
  ⠸ exec
    ⠙ preparing to launch process
⠼ mirrord cli starting...
  ⠼ exec
    ⠹ preparing to launch process
⠴ mirrord cli starting...
  ⠴ exec
    ⠸ preparing to launch process
⠦ mirrord cli starting...
  ⠦ exec
    ⠼ preparing to launch process
⠧ mirrord cli starting...
  ⠧ exec
    ⠴ preparing to launch process
⠇ mirrord cli starting...
  ⠇ exec
    ⠦ preparing to launch process
⠏ mirrord cli starting...
  ⠏ exec
    ⠧ preparing to launch process
⠋ mirrord cli starting...
  ⠋ exec
    ⠇ preparing to launch process
⠙ mirrord cli starting...
  ⠙ exec
    ⠏ preparing to launch process
⠹ mirrord cli starting...
  ⠹ exec
    ⠋ preparing to launch process
⠸ mirrord cli starting...
  ⠸ exec
    ⠙ preparing to launch process
⠼ mirrord cli starting...
  ⠼ exec
    ⠹ preparing to launch process
⠴ mirrord cli starting...
  ⠴ exec
    ⠸ preparing to launch process
⠦ mirrord cli starting...
  ⠦ exec
    ⠼ preparing to launch process
⠧ mirrord cli starting...
  ⠧ exec
    ⠴ preparing to launch process
⠇ mirrord cli starting...
  ⠇ exec
    ⠦ preparing to launch process
⠏ mirrord cli starting...
  ⠏ exec
    ⠧ preparing to launch process
⠋ mirrord cli starting...
  ⠋ exec
    ⠇ preparing to launch process
⠙ mirrord cli starting...
  ⠙ exec
    ⠏ preparing to launch process
⠹ mirrord cli starting...
  ⠹ exec
    ⠋ preparing to launch process
⠸ mirrord cli starting...
  ⠸ exec
    ⠙ preparing to launch process
⠼ mirrord cli starting...
  ⠼ exec
    ⠹ preparing to launch process
⠴ mirrord cli starting...
  ⠴ exec
    x preparing to launch process
⠦ mirrord cli starting...
  x exec
    x preparing to launch process
    x checking for warnings
      ✓ When targeting multi-pod deployments, mirrord impersonates the first pod in the deployment.
 Support for multi-pod impersonation requires the mirrord operator, which is part of mirrord for Teams.
 To try it out, join the waitlist with `mirrord waitlist <email address>`, or at this link: https://metalbear.co/#waitlist-form
      ✓ WARNING: mirrord is running on openshift, due to default PSP of openshift, mirrord may not be able to create the agent. Please refer to https://mirrord.dev/docs/overview/faq/
    ✓ layer extracted
    ✓ no operator detected
    ✓ agent pod created
    ✓ pod is ready
Error:   × Invalid message received from agent "None"
  help: This is a bug. Please report it in our Discord or GitHub repository.
        
        - If you're still stuck:
        
        >> Please open a new bug report at https://github.com/metalbear-co/mirrord/issues/new/choose
        
        >> Or join our Discord https://discord.gg/metalbear and request help in #mirrord-help
        
        >> Or email us at hi@metalbear.co

```

**Comment by infiniteregrets** [comment]
blocked by https://github.com/tamasfe/termspin/issues/1

**Comment by aviramha** [comment]
Fixed in #1781
